libs/components/core/src/lib/modules/resize-observer/resize-observer-media-query.service.ts

```diff
@@ -122,11 +130,38 @@ export class SkyResizeObserverMediaQueryService implements OnDestroy {
       });
   }
 
-  #updateBreakpoint(breakpoint: SkyMediaBreakpoints) {
+  #updateBreakpoint(
+    breakpoint: SkyMediaBreakpoints,
+    updateResponsiveClasses?: boolean
+  ): void {
+    if (updateResponsiveClasses) {
+      this.#updateResponsiveClasses(this.#currentBreakpoint, breakpoint);
+    }
+
     this.#currentBreakpoint = breakpoint;
     this.#currentBreakpointObs.next(breakpoint);
   }
 
+  #updateResponsiveClasses(
+    oldBreakpoint: SkyMediaBreakpoints,
+    newBreakpoint: SkyMediaBreakpoints
+  ): void {
+    const oldClass = `sky-responsive-container-${
+      this.#breakpoints.find((breakpoint) => breakpoint.name === oldBreakpoint)
+        ?.classSuffix
+    }`;
+    const newClass = `sky-responsive-container-${
+      this.#breakpoints.find((breakpoint) => breakpoint.name === newBreakpoint)
+        ?.classSuffix
+    }`;
+
+    if (this.#target?.nativeElement.classList.contains(oldClass)) {
```

Comment:
`this.#target.nativeElement.classList` is repeated twice, so assign it to a const. There's also no need to check for the existence of the class when removing since it will not result in an error if it's not in the list.

```
const targetClassList = this.#target?.nativeElement.classList;

targetClassList?.remove(oldClass);
targetClassList?.add(newClass);
```

Alternatively you could inject `Renderer2` and use its methods for adding and removing classes. That's Angular's recommended method for interacting with the DOM in these types of scenarios.

```
#renderer = inject(Renderer2);
...
#renderer.removeClass(this.#target, oldClass);
#renderer.addClass(this.#target, newClass);
```